dla_cnn/desi/training_sets.py

```diff
@@ -111,3 +112,154 @@ def prepare_training_test_set(ids_train, ids_test,
+    -------------------------------------------------------------------------------
+    return:
+    nhisample:list
+    zsample:list
+    sampleid:sightline idlist
+    sample_dla_id:np.ndarray
+    """
+    
+    nhi=[]
+    zdla=[]
+    dlaid=[]
+    mockid=[]
+    for ii in idlist:
+        sightline=sightline_retriever(ii,Mocks)
+        for jj in sightline.dlas:
+            if 912<=(jj.central_wavelength/(1+sightline.z_qso))<=1220:#dla restframe[912,1220]
+                nhi.append(jj.col_density) 
+                zdla.append(float(jj.central_wavelength/1215.6701-1))
+                mockid.append(ii)
+                dlaid.append(str(ii)+jj.id)
+    #cut NHI(20.3-21.5),zdla with rsd,dlaid
+    nhicut=(np.array(nhi)>=20.3)&(np.array(nhi)<=21.2)&(np.array(zdla)>=2.13)#dla central wavelength>3800
+    nhi_dla=np.array(nhi)[nhicut]
+    zdla=np.array(zdla)[nhicut]
+    mockid=np.array(mockid)[nhicut]
+    dlaid=np.array(dlaid)[nhicut]
+    #z split bins, get index
+    z_index=split_bins(zdla,z_bins)
+    #nhi split bins,get index
+    nhi_index=[]
+    for ii in range(0,z_bins):
+        nhi_index.append(split_bins(nhi_dla[z_index[ii]],nhi_bins))
+    #get bins id
+    dlaid_bins=[]
+    for i in range(0,z_bins):
+        for j in range(0,nhi_bins):
+            dlaid_bins.append(dlaid[z_index[i]][nhi_index[i][j]])
+    #uniform idsample in every bins
+    sample_dla_id=[]
+    for jj in range(0,len(dlaid_bins)):
+        sample_dla_id.append(np.random.choice(list(dlaid_bins[jj]),num))
+    sample_dla_id=np.array(sample_dla_id).ravel()
+    nhisample=[]
+    zsample=[]
+    sampleid=[]
+    for i in range(0,len(sample_dla_id)): 
+        index=np.where(dlaid==sample_dla_id[i])
+        nhisample.append(nhi_dla[index])
+        zsample.append(zdla[index])
+        sampleid.append(mockid[index])
+    #return
+    return nhisample,zsample,sampleid,sample_dla_id
+
+def select_samples_50p_pos_neg(sightline,kernel):
+    """
+    For a given sightline, generate the indices for DLAs and for without
+    Split 50/50 to have equal representation
+    Parameters
+    ----------
+    sightline:dla_cnn.data_model.Sightline.Sightline object
+    kernel:int
+    Returns
+    -------
+    idx: np.ndarray
+        positive + negative indices
+    """
+  
+    lam, lam_rest, ix_dla_range = get_lam_data(sightline.loglam, sightline.z_qso, REST_RANGE=[900,1346])
+    kernelrangepx = int(kernel/2)
+    cut=((np.nonzero(ix_dla_range)[0])>=kernelrangepx)&((np.nonzero(ix_dla_range)[0])<=(len(lam)-kernelrangepx-1))
+    newclassification=sightline.classification[cut]
+    num_pos = np.sum(newclassification==1, dtype=np.float64)
+    num_neg = np.sum(newclassification==0, dtype=np.float64)
+    n_samples = int(min(num_pos, num_neg))
+
+    r = np.random.permutation(len(newclassification))
+
+    pos_ixs = r[newclassification[r]==1][0:n_samples]
+    neg_ixs = r[newclassification[r]==0][0:n_samples]
+    # num_total = data[0].shape[0]
```

Comment:
Remove the commented code if it isn't needed.

Reply:
ok